pkg/cluster/kafkaservice/kafkaservice.go

```diff
@@ -0,0 +1,65 @@
+package kafkaservice
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/redhat-developer/app-services-cli/pkg/api/kas"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/constants/kafka"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/utils"
+	"github.com/redhat-developer/app-services-cli/pkg/kafka/kafkaerr"
+)
+
+type KafkaConnection struct {
+}
+
+func (k *KafkaConnection) CustomResourceExists(ctx context.Context, c *cluster.KubernetesCluster, namespace string, serviceName string, opts cluster.Options) error {
+
+	path := kafka.GetKafkaConnectionsAPIURL(namespace)
+
+	err := utils.ResourceExists(ctx, c, path, serviceName, opts)
+
+	return err
+}
+
+func (k *KafkaConnection) CreateCustomResource(ctx context.Context, c *cluster.KubernetesCluster, serviceID string, namespace string, opts cluster.Options) error {
+
+	api := opts.Connection.API()
+
+	path := kafka.GetKafkaConnectionsAPIURL(namespace)
+
+	kafkaInstance, _, err := api.Kafka().GetKafkaById(ctx, serviceID).Execute()
+	if kas.IsErr(err, kas.ErrorNotFound) {
+		return kafkaerr.NotFoundByIDError(serviceID)
+	}
+
+	serviceName := kafkaInstance.GetName()
+
+	kafkaConnectionCR := kafka.CreateKCObject(serviceName, namespace, serviceID)
+
+	crJSON, err := json.Marshal(kafkaConnectionCR)
+	if err != nil {
+		return fmt.Errorf("%v: %w", opts.Localizer.MustLocalize("cluster.kubernetes.createKafkaCR.error.marshalError"), err)
+	}
+
+	resource := kafka.AKCResource
+
+	err = utils.CreateResource(ctx, c, path, serviceName, namespace, crJSON, resource, opts, GetWatchErrorMessages())
+
+	return err
+}
+
+func GetWatchErrorMessages() map[string]string {
```

Comment:
[Question]

What is purpose of that function - do we need function for something that should be constant?
Is there any benefit for having slice vs explicit error strings (having this this way will affect our linter capability)
Best to avoid it unless there is reason for it

Reply:
Oh.. this is not even used in this file, which is also broken responsibility. Keys should be defined as close to the actual usage.